plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/core/MavenWorkspace.java

```diff
@@ -110,9 +110,16 @@ public void projectUpdated(Map<MavenProject, MavenProjectModifications> updated,
                 List<MavenProject> needResolve = manager.findDependentProjects(allChangedProjects);
                 needResolve.addAll(updated.keySet());
 
+                List<String> updatedPaths = updated.keySet().stream()
```

Comment:
Sorry, but since when did "streaming ideology" become an argument for code quality? The version you propose has the advantage that it is easier to debug (because I can use line breakpoints), on the other hand, it will perform worse and is longer.
Btw: if we want to be more catholic than the pope, it would have to be `.map(project::getProject)`, etc.